spec/models/post_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { Post.new(title: 'Javascript', text: 'porgramming language', comments_counter: 10, likes_counter: 10) }
+  before { subject.save }
+
+  it 'The title should be present' do
+    subject.title.nil?
+    expect(subject).to_not be_valid
+  end
+
+  it 'The title length should not be exceed more than 250 characters' do
+    subject.title = 'J' * 280
+    expect(subject).to_not be_valid
+  end
+
+  it 'The comments_counter should be integer' do
+    subject.comments_counter = 'b'
+    expect(subject).to_not be_valid
+  end
+
+  it 'The comments_counter should be greater than or equal to zero' do
+    subject.comments_counter = -8
+    expect(subject).to_not be_valid
+  end
+
+  it ' Should test comments_counter to be >= 0' do
+    expect(subject.comments_counter).to be >= 0
+  end
+
+  it 'The likes_counter should be integer' do
+    subject.likes_counter = 'c'
+    expect(subject).to_not be_valid
+  end
+
+  it 'The likes_counter should be greater than or equal to zero' do
+    subject.likes_counter = 2.5
+    expect(subject).to_not be_valid
+  end
+
+  it ' Should test likes_counter to be >= 0' do
+    expect(subject.likes_counter).to be >= 0
+  end
+
+  it 'can update posts counter' do
+    expect(subject).to respond_to(:update_posts_counter)
+  end
+
+  it 'can display most recent comments' do
+    expect(subject).to respond_to(:most_recent_comments)
+  end
```

Comment:
- [ ] Good job adding tests that show the post object responds to the methods. However, you need to add more comprehensive tests that demonstrate the methods behave as expected. Kindly add tests (or modify these ones):

  - that shows that the `update_posts_counter` actually updates the posts counter of the post author.
  - that the `most_recent_comments` method fetches the recent comments of the post object. You may need to create comments in this file and ensure the expected comments match the ones you created.